### DISCUSSION TITLE
How can I add a global check registration for app_commands?

### DISCUSSION THREAD

**Discussion by Ibrahim2750mi** [Rapptz/discord.py#9293] [Q&A] [answered]
There are [functions](https://github.com/Rapptz/discord.py/blob/master/discord/ext/commands/bot.py#L350) dedicated in `ext.commands.Bot` to add global checks, is there any method to do so for `app_commands`?

**Comment by ika2kki** [comment, marked as answer]
Hi Ibrahim,

You can subclass `CommandTree` and override the `interaction_check` method, which behaves as a global check for all commands.
For example,
```py
import discord
from discord import app_commands

class MyCommandTree(app_commands.CommandTree):
    async def interaction_check(self, interaction: discord.Interaction) -> bool:
        ...
```

If you're using `ext.commands.Bot` as the client instance, you can attach this tree to the bot via the `tree_cls` keyword argument in the bot constructor:
```py
bot = commands.Bot(
    command_prefix="?",
    intents=discord.Intents.default(),
    tree_cls=MyCommandTree
)
```

**Reply by Ibrahim2750mi** [reply]
This looks good enough, but is this the only way?

**Reply by ika2kki** [reply]
> This looks good enough, but is this the only way?

It's the only method called globally, ya.

**Reply by Soheab** [reply]
There is also the recently added [`interaction_check`](https://discordpy.readthedocs.io/en/latest/ext/commands/api.html#discord.ext.commands.Cog.interaction_check) for a local cog check but I don't think that's what you're looking for.

**Reply by Ibrahim2750mi** [reply]
@Soheab will this ever be added to the bot itself?

**Reply by Soheab** [reply]
> @Soheab will this ever be added to the bot itself?

Probably not. The above solution makes sense.
